0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1201 @@
+| `id`                      | [URL][] | _(Server-provided)_ The primary, unique identifier for this account. MUST be an HTTP(S) [URL][] where the client can [get this resource][Get Account]. |
+| `balance`                 | String  | _(Server-provided)_ Balance as decimal [amount][]. Defaults to `"0"`. This can be negative, if the account's `minimum_allowed_balance` allows it. |
+
+A ledger MAY define additional fields for the account resource. For example, `is_admin` could be a boolean that, if true, grants administrator-level authorization when authenticated as the account owner.
+
+
+### Message Resource
+[Message resource]: #message-resource
+
+Messages are sent through the ledger's [Send Message][] method and received in a [WebSocket][] subscription. All fields of the message are required:
+
+| Field    | Value   | Description                                             |
+|:---------|:--------|:--------------------------------------------------------|
+| `ledger` | [URL][] | The base [URL][] of this ledger. MUST be an HTTP(S) URL where the client can [Get Ledger Metadata][]. |
+| `from`   | [URL][] | Resource identifier of the account sending the message. MUST be an HTTP(S) URL where the client can [get account information][Get Account]. The sender of a message MUST be authenticated as the owner of this account. |
+| `to`     | [URL][] | Resource identifier of the account receiving the message. MUST be an HTTP(S) URL where the client can [get account information][Get Account]. |
+| `data`   | Object  | The message to send, containing arbitrary data. A ledger MAY set a maximum length on messages, but that limit MUST NOT be less than 510 UTF-8 characters or 2,048 bytes. |
+
+
+### Asset Resource
+[Asset resource]: #asset-resource
+
+There are many kinds of resources that can be tracked in a ledger. The Asset Resource describes the resource tracked in this ledger, so that client applications can display appropriate information to users. An asset resource has the following fields:
+
+| Field            | Value     | Description                                   |
+|:-----------------|:----------|:----------------------------------------------|
+| `type`           | String    | The type of asset. Currently, the only supported type is `iso4217-currency`. |
+| `code`           | String    | The currency code to represent this asset. For `iso4217-currency` assets, this MUST be a three-letter uppercase [ISO 4217](http://www.xe.com/iso4217.php) currency code. |
+| `symbol`         | String    | Symbol to use in user interfaces with amounts of this asset. For example, "$". |
+| `decimal_digits` | Number    | _(Optional)_ Typical number of decimal places to display for amounts of this asset. MUST be a non-negative integer. This MAY be different from the precision and scale used internally by the ledger, which are reported in the [Get Ledger Metadata][] field. |
```

Comment:
The description for this field could be improved. `decimal_digits` refers to the **conventionally displayed** number of decimal places for a given asset. It's purely a hint for display purposes and is ignored by everyone but human-facing amount formatters. (By contrast, precision and scale affect that actual available precision on a ledger.)

For example, a USD ledger might use a precision of 15 and a scale of 6. So, it can represent amounts like USD 10.123456. However, US dollars are typically shown with a precision of cents, so `decimal_digits` would be 2. How a client exactly renders this is up to the client, but it might show any precision beyond the `decimal_digits` in a deemphasized way, e.g. 10.12(3456)